examples/drawing/plot_mpl_annotations.py

```diff
@@ -0,0 +1,72 @@
+"""
+=================
+Annotations with Matplotlib
+=================
+
+Example of using matplotlib to place annotations on top of a network
+"""
+
+import matplotlib.pyplot as plt
+import matplotlib.image as mpimg
+import networkx as nx
+import urllib.request
+import io
+
+# Image URLs for graph nodes
+URLs = [
+    {
+        "type": "router",
+        "URL": "https://www.materialui.co/materialIcons/hardware/router_black_144x144.png",
+    },
+    {
+        "type": "switch",
+        "URL": "https://www.materialui.co/materialIcons/action/dns_black_144x144.png",
+    },
+    {
+        "type": "PC",
+        "URL": "https://www.materialui.co/materialIcons/hardware/computer_black_144x144.png",
+    },
+]
+
+# Import images from URLS into a collection
+images = {}
+for u in URLs:
+    with urllib.request.urlopen(u["URL"]) as url:
+        image = mpimg.imread(io.BytesIO(url.read()))
+        images[u["type"]] = image
+
+# Generate the computer network graph
+G = nx.Graph()
+
+G.add_node("router", image=images["router"])
+for i in range(1, 4):
+    G.add_node("switch_" + str(i), image=images["switch"])
+    for j in range(1, 4):
+        G.add_node("PC_" + str(i) + "_" + str(j), image=images["PC"])
+
+G.add_edge("router", "switch_1")
+G.add_edge("router", "switch_2")
+G.add_edge("router", "switch_3")
+for u in range(1, 4):
+    for v in range(1, 4):
+        G.add_edge("switch_" + str(u), "PC_" + str(u) + "_" + str(v))
+
+pos = nx.spring_layout(G)
+fig, ax = plt.subplots()
+nx.draw_networkx_edges(G, pos=pos, ax=ax)
+
+tr_figure = ax.transData.transform
+tr_axes = fig.transFigure.inverted().transform
+
+# Draw the images on top of the graph
+icon_size = 0.05
+icon_center = icon_size / 2.0
```

Comment:
This is an important step, so it'd be nice to have a good comment here explaining that the icon size is being set in terms of axis dimensions, and transform objects are being created to map from figure/axes space back to data space.

Reply:
I'm not sure, but it seems like the transforms can be consolidated. Is there a reason why both `transFigure` and `transData` are needed?

Reply:
> I'm not sure, but it seems like the transforms can be consolidated. Is there a reason why both `transFigure` and `transData` are needed?

We see ever to create overlapped axes for placing each icon, which requires xlim-ylim scaling. Then we need to place the icon with reference to the main axes which requires a 0-1 scaling.
We will look into if this can be done in just 3 lines instead of 5 but we have documented it well right now either way :)

Reply:
Agreed - it certainly gets the job done :)